subprojects/ce-login/celogin/src/CeLoginUtil.cpp

```diff
@@ -305,28 +329,49 @@ CeLogin::CeLoginRc CeLogin::decodeAndVerifySignature(
     // Verify signature over SourceFileData
     if (CeLoginRc::Success == sRc)
     {
-        // Use the Digest NID without the signature algorithm (i.e. Sha512 vs
-        // Sha512WithRSAEncryption).
-
-        // returns 1 on successful verification
-        int sRsaResult = RSA_verify(
-            CeLogin_Digest_NID, sHashReceivedJson, sizeof(sHashReceivedJson),
-            decodedAsnParm->signature->data, decodedAsnParm->signature->length,
-            sPublicKey);
-        if (1 != sRsaResult)
+        int sVerifyResult = 1;
+        EVP_PKEY_CTX* sCtx = EVP_PKEY_CTX_new(sPublicKey, NULL /* no engine */);
```

Comment:
Consider wrapping the EVP_PKEY_verify logic in a function so that it can be reused in both locations without copy/paste